### PR TITLE
 feat(model): class with parameter fitting 

### PR DESCRIPTION
**Pull request by felipehuerta17**
ODE systems embedded into classes, the class model represent "SIR" like models. SIR and SIR-X derive functions from Model.

model_ now receives time vectors instead of final time

Model.fit() method was included to fit the parameter alpha of SIR and SIR-X models to observed data of number of infected.

Work done in a 3.5h video call with @jia200x 

Jose at 12 pm programming

### CODE REVIEW

File: SIR-X.ipynb [hunk 1]
```diff
@@ -240,6 +243,268 @@
...
...jowPBDNbaWY7zazazL4adj3Dwcz2mlmVmW0xs7T8tj8zu9/MGsxsW59lxWb2tJm9EfycHGaNJ2KQ/txuZm8Hn9MWM/tImDWeCDObbWbPmdlrZrbdzP48WJ7On9FgfUrLz8nM8s3sZTN7NejPt4LlpWa2MfiMfho8YiC5bWbyISMzywJ2AVeSeFjPJuB6d0/bZzdDIhCAcndP2+unzewyoA34kbsvCZb9E9Dk7v8YhPdkd/+fYdaZrEH6czvQ5u7fDrO2k2FmM4AZ7v6KmRUCm4FrgD8mfT+jwfr0KdLwcwqeW1/g7m1mlgP8Fvhz4MvAz939ETP7N+BVd78nmW1m+ghhGVDt7nvcvRt4BFgVck0CuPvzJJ6d0dcq4IFg+gES/1jTwiD9SVvuXu/urwTTrcBrJJ6Hns6f0WB9Skue0BbM5gQvBz4APBosP6HPKNMDYSZQ02e+ljT+H6APB/7DzDYHz5zOFNPdvR4S/3iBaSHXMxxuNbOtwSGltDm80peZzQPOBzaSIZ9Rvz5Bmn5OZpZlZluABuBpYDdw2N2jQZMT2udleiAM9IT5TDhGdom7XwB8GPjT4HCFpJ57gAXAe4B64DvhlnPizGwC8BjwJXdvCbue4TBAn9L2c3L3mLu/h8Rz6ZcBZw3ULNntZXog1AKz+8zPAupCqmXYuHtd8LMBeJzE/wiZ4EBwnLf3eG9DyPWcEnc/EPyDjQP/Tpp9TsFx6ceAH7v7z4PFaf0ZDdSndP+cANz9MLAeuAiYZGa9j0c+oX1epgfCJqAsOOueS+JZzxUh13RKzKwgOCGGmRUAHwK2HX+ttFEB3BhM3wg8EWItp6x3xxn4OGn0OQUnLO8DXnP3/9PnrbT9jAbrU7p+TmZWYmaTgulxwAdJnBd5Drg2aHZCn1FGX2UEEFxC9i9AFnC/u/+vkEs6JWY2n8SoACAbeDgd+2RmPwFWkPhmxgPAN4FfAGuAOcBbwHXunhYnagfpzwoShyEc2AulhJHYAAAAdklEQVR8rvf4e6ozs0uB3wBVQDxY/Nckjrmn62c0WJ+uJw0/JzM7l8RJ4ywSf9yvcfc7gn3EI0Ax8HvgM+7eldQ2Mz0QREQkOZl+yEhERJKkQBAREUCBICIiAQWCiIgACgQREQkoEEREBFAgiIhIQIEgIiIA/H/pKn0pHHqzxAAAAABJRU5ErkJggg==\n",
```
Comment (sasalatart) [inline SIR-X.ipynb]: Is this an image generated by the notebook?

Comment (sasalatart) [reply SIR-X.ipynb]: Not sure if we should be version controlling this. I am not familiar with this haha, but I know that base64 images are heavy. Should we also version control the results of executing the notebook?

Comment (felipehuerta17) [reply SIR-X.ipynb]: Output removed!

**Comment by jia200x** [comment]
MAJESTIC

**Comment by felipehuerta17** [comment]
me

**Pull request merged**